### PR TITLE
Add `sublabel` and `size` props to MenuItem and SelectOption

### LINKED ISSUE #206: MenuItem: multiple lines
#### Feature

Multiple lines in `MenuItem` ("sub label" / "description")

image

Should also work in dependent components that use `MenuItem` internally, i.e. `Select`.

---

<div type='discussions-op-text'>

<sup>Originally posted by **rseydler** June 29, 2021 in #179</sup>

I am using the MenuItem to return a list of Bentley connected projects. Due to the length of the project name + project description it will not fit in the MenuItem dropdown. I was hoping to split the dropdown items into two separate lines, one line for project name, then project description on the next line.
image

</div>

### PR DESCRIPTION
**Pull request by mayank99**
- Added new `sublabel` and `size` props to MenuItem and SelectOption.
	- Closes #206
	- Added new stories, and updated tests.

image
image

## Checklist

- [x] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [x] Add component features demo in Storybook (different stories)
- [x] Approve test images for new stories
- [x] Add an entry to the changelog for any new components and changes
- [x] Add screenshots of the key elements of the component
- [x] Add any additional comments describing the features you've implemented

### CODE REVIEW

File: src/core/Menu/MenuItem.tsx [hunk 1]
```diff
@@ -20,13 +20,21 @@ export type MenuItemProps = {
...
    * Callback function that handles click and keyboard submit actions.
    */
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  onClick?: (value?: any) => void;
+  onClick?: (value?: unknown) => void;
+  /**
+   * Modify height of the item. Use 'large' to support multiple lines.
+   * @default 'default'
+   */
+  size?: 'default' | 'large';
```
Comment (veekeys) [inline src/core/Menu/MenuItem.tsx:32]: Is it useful without sublabels?

File: src/core/Menu/MenuItem.tsx [hunk 2]
```diff
@@ -103,7 +114,10 @@ export const MenuItem = React.forwardRef<HTMLLIElement, MenuItemProps>(
           React.cloneElement(icon, {
             className: cx(icon.props.className, 'iui-icon'),
           })}
-        <span className='iui-content'>{children}</span>
+        <span className='iui-content'>
+          <div className='iui-menu-label'>{children}</div>
+          {sublabel && <div className='iui-menu-description'>{sublabel}</div>}
```
Comment (veekeys) [inline src/core/Menu/MenuItem.tsx:119]: There is a limitation from css, that if it is not large, sublabel wont shown? Cause you do not check it in code here

File: src/core/Select/Select.tsx [hunk 3]
```diff
@@ -237,14 +245,14 @@ export const Select = <T,>(props: SelectProps<T>): JSX.Element => {
...
           disabled: option.disabled,
-          value: option.value,
           onClick: () => {
             !option.disabled && onChange?.(option.value);
             close();
           },
           ref: (el: HTMLElement) => isSelected && el?.scrollIntoView(),
           icon: option.icon,
           role: 'option',
+          ...option,
```
Comment (veekeys) [inline src/core/Select/Select.tsx:255]: didnt want to add props? so then maybe remove all above.. idk..

File: stories/core/DropdownMenu.stories.tsx [hunk 4]
```diff
@@ -4,13 +4,20 @@
...
+  IconButton,
+  MenuItem,
+} from '../../src/core';
+import {
+  SvgClipboard,
+  SvgCrop,
+  SvgMore,
+  SvgMove,
+  SvgPlaceholder,
+} from '@itwin/itwinui-icons-react';
```
Comment (veekeys) [inline stories/core/DropdownMenu.stories.tsx:20]: all my icons work ;(

File: src/core/Select/Select.tsx [hunk 5]
```diff
@@ -28,6 +28,14 @@ export type SelectOption<T> = {
    * Label of the item used in dropdown list and when selected.
    */
   label: string;
+  /**
+   * Sublabel of the item shown below the label when item size is 'large'.
+   */
+  sublabel?: React.ReactNode;
```
Comment (veekeys) [inline src/core/Select/Select.tsx:34]: tests

File: src/core/Menu/MenuItem.tsx (hunk 2 above)
Comment (mayank99) [reply src/core/Menu/MenuItem.tsx:119]: Yeah, it's set in css

File: src/core/Menu/MenuItem.tsx (hunk 1 above)
Comment (mayank99) [reply src/core/Menu/MenuItem.tsx:32]: Can be. Maybe with custom renderer or something

File: stories/core/DropdownMenu.stories.tsx (hunk 4 above)
Comment (mayank99) [reply stories/core/DropdownMenu.stories.tsx:20]: This is stories 😄

File: src/core/Menu/MenuItem.tsx (hunk 1 above)
Comment (veekeys) [reply src/core/Menu/MenuItem.tsx:32]: I find it weird...
Cant we just have sublabels and toggle that size for them?

File: src/core/Select/Select.tsx (hunk 3 above)
Comment (mayank99) [reply src/core/Select/Select.tsx:253]: Removed all

File: src/core/Select/Select.tsx (hunk 5 above)
Comment (mayank99) [reply src/core/Select/Select.tsx:34]: Added

File: src/core/Menu/MenuItem.tsx (hunk 1 above)
Comment (mayank99) [reply src/core/Menu/MenuItem.tsx:33]: Defaulted to 'large' when sublabel is passed

File: src/core/Select/Select.tsx (hunk 5 above)
Comment (veekeys) [reply src/core/Select/Select.tsx:34]: Add test for the case, when size not passed, but has sublabel

Comment (veekeys) [reply src/core/Select/Select.tsx:34]: Approving, so you could merge when test is there

**Review by veekeys — APPROVED** [review]

**Pull request merged**